### PR TITLE
Initial implementation of News page

### PR DESCRIPTION
**Pull request by CheddarChoi**
Implemented
- Linear view of the news item
- Modal when clicking the "Click here to see the details" button
- Tags (open/closed) based on end date

Not implemented yet
- Responsive view (mobile version)

Capture-2024-02-29-145727
Capture-2024-02-29-145757

### CODE REVIEW

File: public/posts/2024-01-25-yoonjoo-award.webp
Comment (alextio) [inline public/posts/2024-01-25-yoonjoo-award.webp:1]: Maybe we should have a subdirectory for image files

File: src/components/NewsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    color: ${Color.black};
+  }
+`
+
+export default function NewsModal({ post, onClose }: { post: Post | null; onClose: () => void }) {
+  const [mdContent, setMdContent] = React.useState<string | null>(null)
+  if (!post) {
+    return null
+  } else {
+    var currentDate = new window.Date()
```
Comment (jhw123) [inline src/components/NewsModal.tsx:65]: don't use `var`. Use `let` or `const`. You use `const` here.

File: src/data/posts.ts [hunk 2]
```diff
@@ -2,7 +2,9 @@ interface Props {
   title: string
   date: Date
   categories: string[]
-  img: string
+  summary: string
+  content?: string // Add path to markdown file if exists
```
Comment (jhw123) [inline src/data/posts.ts:6]: can we make the property name more evident? e.g., `contentMdFilePath`

File: src/components/NewsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                  </Category>
+                ))}
+                {closed && <Category style={{ backgroundColor: Color.gray400 }}>Closed</Category>}
+                {open && <Category style={{ backgroundColor: Color.green300 }}>Open</Category>}
+              </CategoryContainer>
+              <Title>{post.title}</Title>
+            </div>
+            <Close onClick={onClose}>
+              <Image src="/images/close.svg" alt="Close" width="24" height="24" />
+              {/* <img width="24" src="/images/close.svg" alt="Close" /> */}
```
Comment (jhw123) [inline src/components/NewsModal.tsx:95]: remove this if not used

File: src/app/news/page.tsx [hunk 4]
```diff
@@ -1,15 +1,52 @@
...
+'use client'
+
+import { useState } from 'react'
+import NewsCard from '@/components/NewsCard'
+import { NewsLine, NewsSection } from '@/components/Section'
+import Post, { POSTS } from '@/data/posts'
+import NewsModal from '@/components/NewsModal'
 
 export default function Page() {
+  const [modal, setModal] = useState(false)
```
Comment (jhw123) [inline src/app/news/page.tsx:10]: do we need this variable? isn't `modalContent === null equivalent` to `modal === false`?

**Review by jhw123 — CHANGES_REQUESTED** [review]

**Comment by jhw123** [comment]
Assign yourself as the assignee for your PR

File: src/app/news/page.tsx (hunk 4 above)
Comment (CheddarChoi) [reply src/app/news/page.tsx]: I removed this in the [second commit](https://github.com/kixlab/website/pull/15/commits/e49beb647bb716fcaafc5caaae734ff75d5c005b)

**Review by jhw123 — APPROVED** [review]

**Review by alextio — APPROVED** [review]

**Pull request merged**